src/basic/copy.c

```diff
@@ -132,6 +130,10 @@ int copy_bytes(int fdf, int fdt, uint64_t max_bytes, bool try_reflink) {
                         assert(max_bytes >= (uint64_t) n);
                         max_bytes -= n;
                 }
+                /* sendfile accepts at most SSIZE_MAX-offset bytes to copy,
+                 * so reduce our maximum by the amount we already copied,
+                 * but always stay above 0. */
+                m = MAX(1u, m - n);
```

Comment:
It's not so easy, because it would be necessary to make sendfile fail. Not impossible, but I don't think it's worth the extra scaffolding that would be necessary.